grr/server/grr_response_server/gui/api_plugins/metadata_test.py

```diff
@@ -0,0 +1,756 @@
+      "GET /metadata_test/method3": {
+        "200": {
+          "description": "The call to the Method3PrimitiveTypes API method "
+                         "succeeded and it returned an instance of "
+                         "BinaryStream.",
+          "content": {
+            "application/octet-stream": {
+              "schema": {
+                "type": "string",
+                "format": "binary"
+              }
+            }
+          }
+        },
+        "default": {
+          "description": "The call to the Method3PrimitiveTypes API method did "
+                         "not succeed."
+        },
+      },
+
+      # Method4RepeatedField (GET, POST) => No result type.
+      "GET /metadata_test/method4": {
+        "200": {
+          "description": "The call to the Method4RepeatedField API method "
+                         "succeeded."
+        },
+        "default": {
+          "description": "The call to the Method4RepeatedField API method did "
+                         "not succeed."
+        }
+      },
+      "POST /metadata_test/method4": {
+        "200": {
+          "description": "The call to the Method4RepeatedField API method "
+                         "succeeded."
+        },
+        "default": {
+          "description": "The call to the Method4RepeatedField API method did "
+                         "not succeed."
+        }
+      },
+    }
+
+    # Extract the responses from the OpenAPI description for the tested routes.
+    open_api_paths_dict = self.open_api_desc_dict["paths"]
+
+    # Get identifiers for each method's OpenAPI dictionary for easier access.
+    method2_path_dict = open_api_paths_dict["/metadata_test/method2"]
+    method3_path_dict = open_api_paths_dict["/metadata_test/method3"]
+    method4_path_dict = open_api_paths_dict["/metadata_test/method4"]
+
+    # Put together the extracted response fields from the OpenAPI description.
+    open_api_responses = {
+      # Method2WithResultType routes and their responses.
+      "GET /metadata_test/method2": method2_path_dict["get"]["responses"],
+      "HEAD /metadata_test/method2": method2_path_dict["head"]["responses"],
+      "POST /metadata_test/method2": method2_path_dict["post"]["responses"],
+
+      # Method3PrimitiveTypes route and its response.
+      "GET /metadata_test/method3": method3_path_dict["get"]["responses"],
+
+      # Method4RepeatedField routes and their responses.
+      "GET /metadata_test/method4": method4_path_dict["get"]["responses"],
+      "POST /metadata_test/method4": method4_path_dict["post"]["responses"],
+    }
+
+    self.assertDictEqual(expected_responses, open_api_responses)
+
+  def testPrimitiveTypesAreCorrectlyDescribedInOpenApiDescription(self):
+    expected_primitives = {
+      "field_double": {"type": "number", "format": "double"},
+      "field_float": {"type": "number", "format": "float"},
+      "field_int64": {"type": "integer", "format": "int64"},
```

Comment:
This came up in a [recent code review](https://github.com/google/grr/pull/814#discussion_r462392915): [Protobuf serializes large integers as string](https://developers.google.com/protocol-buffers/docs/proto3), because [JavaScript loses precision](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER) around `2^53-1`. Is this relevant for the OpenAPI description? Should we use `"type": "string", "format": "int64"`?

Reply:
Thank you very much for pointing this out, Max! I believe this is relevant for the OpenAPI description because, for example, it should let clients know what types to expect in the result JSON.

Reply:
We discussed about this and the answer was that the behavior is the same. I made the changes where applicable. Thank you @max-vogler and @mbushkov !